services/scan/src/central_scanner_app.ts

```diff
@@ -472,7 +473,57 @@ export async function buildCentralScannerApp({
         return;
       }
 
-      response.type('text/plain; charset=utf-8');
+      const parseBodyResult = safeParse(Scan.ExportRequestSchema, request.body);
+      if (parseBodyResult.isErr()) {
+        response.status(400).json({
+          status: 'error',
+          errors: [
+            {
+              type: 'invalid-request',
+              message: parseBodyResult.err().message,
+            },
+          ],
+        });
+        return;
+      }
+
+      const { directoryPath, filename } = request.body;
+
+      await fs.mkdir(directoryPath, { recursive: true });
+      const file = await fs.open(path.join(directoryPath, filename), 'w');
+      const fileStream = file.createWriteStream();
```

Comment:
This doesn't appear to restrict what directories it is allowed to write to. Did you see [`exportToUsbDrive`](https://github.com/votingworks/vxsuite/blob/b671e57f4e5934256fe20e706ada886192f7ab3b/libs/data/src/exporter.ts#L124)? That ensures the path is one that is allowed to be written to in case the client sends a malicious request. Check out the [`/central-scanner/scan/backup-to-usb-drive` endpoint](https://github.com/votingworks/vxsuite/blob/b671e57f4e5934256fe20e706ada886192f7ab3b/services/scan/src/central_scanner_app.ts#L615-L630). Also, note that I renamed that endpoint because I explicitly didn't want the client to be allowed to choose where it went. Maybe do that here, too?

Reply:
Ah, thanks for the pointers! Hadn't looked at the backup endpoint - will update to leverage `exportToUsbDrive`